src/views/Variables/ListView.vue

```diff
@@ -111,4 +116,16 @@
       }
     ]
   })
+
+  const checkIfIsEditable = (item) => {
+    enableRedirect.value = !item.value.isSecret
+  }
+
+  onBeforeRouteLeave((to, from, next) => {
+    if (enableRedirect.value) {
+      return next()
+    }
+    enableRedirect.value = true
+    return next(false)
```

Comment:
maybe you can use the variable directly in the next() function:

return next(enableRedirect.value)

Reply:
Nesse caso eu preciso setar a variavel como true antes de bloquear com next(false), para que, caso eu selecione algo que me redirecione par outra tela (exemplo, botado de criar uma nova variavel) o user não seja bloqueado